tests/test_connection.py

```diff
@@ -61,7 +73,8 @@ def test_connect_uri(self):
         c.admin.authenticate("admin", "password")
         c.mongoenginetest.add_user("username", "password")
 
-        self.assertRaises(ConnectionError, connect, "testdb_uri_bad", host='mongodb://test:password@localhost')
```

Comment:
Might be better to move the bad uri tests into a single test case and then ignore if pymongo 3+